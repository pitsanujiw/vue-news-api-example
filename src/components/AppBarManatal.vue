<template>
  <v-app-bar app class="elevation-0" dark>
    <div class="text-h5" v-on:click="redirectHome">Manatal News</div>
    <v-spacer></v-spacer>

    <filter-dialog />
    <v-menu left bottom>
      <template v-slot:activator="{ on, attrs }">
        <v-btn icon v-bind="attrs" v-on="on">
          <v-icon>mdi-dots-vertical</v-icon>
        </v-btn>
      </template>

      <v-list>
        <v-list-item to="/">
          <v-list-item-title>News</v-list-item-title>
        </v-list-item>
        <v-list-item to="/history">
          <v-list-item-title>History</v-list-item-title>
        </v-list-item>
        <v-list-item @click="setError">
          <v-list-item-title>Try Error Message</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
  </v-app-bar>
</template>

<script>
import { mapActions } from 'vuex';

import FilterDialog from '@/components/FilterDialog.vue';

export default {
  name: 'AppBarManatal',
  components: {
    FilterDialog,
  },
  methods: {
    ...mapActions(['getTopHighLightsError']),
    async setError() {
      await this.getTopHighLightsError(); // for try error page
    },
    redirectHome() {
      this.$router.push('/'); // push router to home page is News page
    },
  },
};
</script>

<style>
</style>
