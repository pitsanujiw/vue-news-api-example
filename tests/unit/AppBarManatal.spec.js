import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import AppBarManatal from '../../src/components/AppBarManatal.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('NewsCard', () => {
  const modules = {
    getters: {},
  };
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store({
      modules: {
        newModules: modules,
      },
    });
  });

  it('Renders AppBarManatal Component', () => {
    const wrapper = shallowMount(AppBarManatal, {
      store,
      localVue,
      vuetify,
      propsData: {
        newsArticle: {
          urlToImage: '__URL__',
          title: '__TITLE__',
          author: '__AUTHOR__',
          description: '__DESCRIPTION__',
          name: '__NAME__',
          source: {
            name: '__SOURCE_NAME__',
          },
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
