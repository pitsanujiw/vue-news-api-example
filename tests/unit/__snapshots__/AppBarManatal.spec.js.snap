// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard Renders AppBarManatal Component 1`] = `
<v-app-bar-stub dark="true" tag="header" extensionheight="48" src="" value="true" app="true" class="elevation-0" newsarticle="[object Object]">
  <div class="text-h5">Manatal News</div>
  <v-spacer-stub tag="div"></v-spacer-stub>
  <filter-dialog-stub></filter-dialog-stub>
  <v-menu-stub opendelay="0" closedelay="0" bottom="true" left="true" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
    <v-list-stub tag="div">
      <v-list-item-stub activeclass="" to="/" tag="div">
        <v-list-item-title-stub tag="div">News</v-list-item-title-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" to="/history" tag="div">
        <v-list-item-title-stub tag="div">History</v-list-item-title-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub tag="div">Try Error Message</v-list-item-title-stub>
      </v-list-item-stub>
    </v-list-stub>
  </v-menu-stub>
</v-app-bar-stub>
`;
